client/http/http.go

```diff
@@ -64,11 +97,16 @@ func (tc *TracedClient) Do(ctx context.Context, req *http.Request) (*http.Respon
 
 	req.Header.Set(correlation.HeaderID, correlation.IDFromContext(ctx))
 
+	start := time.Now()
```

Comment:
@mantzas @tpaschalis Possible `panic` on line 116?
Note how we may receive an error and a `nil` response on line 101, and then attempt to set `rsp.Body` on line 116 if the encoding header is set?

If so, maybe we can tackle as part of this PR.

Let me know if I'm missing something.

Reply:
Αs we discussed, we will take care of this case in this PR, since it's a very small case.

A nil response may be returned either due to a closed circuit breaker, or an error in the `do` method (eg. a nil URL).